portfolio/src/main/webapp/script.js

```diff
@@ -308,113 +323,109 @@ fetchReportMarkers() {
+        destination: new google.maps.LatLng(destLat, destLng),
+        provideRouteAlternatives: true,
+        travelMode: 'DRIVING'
     }
-  });
-  console.log('2222222');
+ 
+    directionsDisplay.setMap(map);
+    directionsService.route(request, function(result, status) {
+        if (status === 'OK') {
+            console.log(result.routes.length);
+ 
+            var counter = 0;
+            var routeIndex = 0;
+            var lessCrimesInRoute = 1000;
+ 
+            for (var r=0; r < result.routes.length; r++){
+            var route = result.routes[r];
+ 
+                for (var j=0; j < route.legs[0].steps.length; j++){
+                    var routeArray = new google.maps.Polyline({
+                    path: [
+                        new google.maps.LatLng(route.legs[0].steps[j].start_location.lat(), route.legs[0].steps[j].start_location.lng()),
+                        new google.maps.LatLng(route.legs[0].steps[j].end_location.lat(), route.legs[0].steps[j].end_location.lng())
+                    ]
+                    });
+                    fetch('/markers').then(response => response.json()).then((markers) => {
+                    markers.forEach((marker) => {
+                    createMarkerForDisplay(marker.lat, marker.lng, marker.crimeType, marker.date, marker.time, marker.description);
+                        var myPosition = new google.maps.LatLng(marker.lat, marker.lng);
+                        if (google.maps.geometry.poly.isLocationOnEdge(myPosition, routeArray, 0.0064)) {
+                            console.log("Crime!");
+                            var a = rateCrime(marker.crimeType);
+                            console.log(a);
+                            counter += a;
+                            console.log(counter);
+                        }
+                        });
+                    });  
+                }
+                console.log(counter);
+                if (counter < lessCrimesInRoute) {
+                    lessCrimesInRoute = counter;
+                    routeIndex = r;
+                    console.log(r);
+                }
+                counter = 0;
+            }
+ 
+            console.log(routeIndex);
+            /*directionsDisplay.setDirections(result);*/
+ 
+            new google.maps.DirectionsRenderer ({
+                map: map,
+                directions: result,
+                routeIndex: routeIndex
+                });
+            }
+    });
 }
-
-function superMath(point1, point2, point3) {
-  console.log(point1.lat());
-  console.log(point1.lng());
-  console.log(point2.lat());
-  console.log(point2.lng());
-  console.log(point3.lat());
-  console.log(point3.lng());
-
-  var a = (point1.lat() - point3.lat()) * (point1.lat() - point3.lat());
-
-  var b = (point1.lng() - point3.lng()) * (point1.lng() - point3.lng());
-
-  var c = Math.sqrt(a + b);
-
-  console.log(c);
+ 
+function rateCrime(crime){
+    console.log(crime);
+    console.log("whats up");
```

Comment:
remove. I'm guessing you needed this because there were too many variables being printed without prompts